### ISSUE TITLE
Address boundary error

### ISSUE THREAD

**Issue by WeirdCarrotMonster** [aio-libs/multidict#6]
Python 3.5.2, multidict-2.0.0 from PyPI, following code fails with SIGSEGV:

```
from multidict import MultiDict
set() - MultiDict().keys()
```

If i understood correctly, error happens in `PyAnySet_Check` call, since set substraction is basically `difference` method call with type check:

```
if (!PyAnySet_Check(so) || !PyAnySet_Check(other)) {
    Py_INCREF(Py_NotImplemented);
    return Py_NotImplemented;
}
return set_difference(so, other);
```

It also can be reproduced with dict and list objects, so i think problem may be related to type checking.

**Comment by asvetlov** [comment]
Good catch!
Would you provide a Pull Request with fix?

**Comment by asvetlov** [comment]
Fixed in both multidict==1.2.2 and multidict==2.0.1
